src/components/intro/index.tsx

```diff
@@ -41,17 +41,31 @@ const demos = [
   },
 ];
 
+const installButtonSource = 'introInstallButton';
+
 interface Props {
   onFile: (file: File | Fileish) => void;
   showSnack: SnackBarElement['showSnackbar'];
 }
 interface State {
   fetchingDemoIndex?: number;
+  installEvent?: BeforeInstallPromptEvent;
 }
 
 export default class Intro extends Component<Props, State> {
   state: State = {};
   private fileInput?: HTMLInputElement;
+  private installingViaButton = false;
```

Comment:
I made this an instance member, rather than state, since it doesn't impact rendering.

Reply:
Squoosh only has one in-app install button, so this works fine. 

The code I used was written to support multiple install buttons on a page, and make it possible to track which button the initiated the install prompt. The idea being each button click handler would set a different identifier in `installSource`. Then when we'd be able to tell the CTR and acceptance of each button, and what specific button (or browser interface) was use used to install the PWA